services/docker/docker-version.service.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { anyInteger } = require('../validators')
+const {
+  buildDockerUrl,
+  getDockerHubUser,
+  getMultiPageData,
+} = require('./docker-helpers')
+const { BaseJsonService } = require('..')
+const { NotFound } = require('..')
+
+const buildSchema = Joi.object({
+  count: anyInteger,
+  results: Joi.array()
+    .items(
+      Joi.object({
+        name: Joi.string(),
+        images: Joi.array().items(
+          Joi.object({
+            digest: Joi.string(),
+            architecture: Joi.string(),
+          })
+        ),
+      }).required()
+    )
+    .required(),
+}).required()
+
+module.exports = class DockerVersion extends BaseJsonService {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return buildDockerUrl('version', true)
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Docker Image Version',
+        pattern: ':user/:repo',
+        namedParams: { user: '_', repo: 'alpine' },
+        staticPreview: this.render({ version: '3.11.2' }),
+      },
+      {
+        title: 'Docker Image Version (tag)',
+        pattern: ':user/:repo/:tag',
+        namedParams: { user: '_', repo: 'alpine', tag: '3.10' },
+        staticPreview: this.render({ version: '3.10.3' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'version' }
+  }
+
+  static render({ version }) {
```

Comment:
We have a lot of helper functions, including for rendering, that handle common scenarios like this. 

https://github.com/badges/shields/blob/master/services/version.js

We can use the `renderVersionBadge` helper instead to avoid duplicating version related items, and it'll also ensure the docker version badges are consistent with the other Shields version badges

Reply:
I noticed that `renderVersionBadge` adds a `v` to versions, this could be misleading as the `v` seems to be stripped in the case of most docker tags.

Reply:
Yes, we should have the `v` prefix on the Version badge. You'll notice this is the paradigm on all our version badges (https://shields.io/category/version) and we need to do the same here. Yes, the value in the badge is not the literal tag you would use for an image pull, but that's also true for some of the other version badges. 

One exception to this would be if the tag value is not a version number, but an arbitrary string. so you could keep the class-level `render` function and if the `version` param is a string (like `next` or `dubnium`, then just return that as-is with the `blue` color, but if `version` is a number then use the `renderVersionBadge` util

Reply:
Looks like the `renderVersionBadge` util won't add the `v` prefix unless it begins with a number anyway. Adjusted.